crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -967,26 +967,27 @@ impl<C: Comments> VisitMut for ServerActions<C> {
             self.fn_decl_ident = old_fn_decl_ident;
         }
 
-        if !self.config.is_react_server_layer {
-            return;
-        }
-
-        let mut child_names = if self.should_track_names {
-            let names = take(&mut self.names);
-            self.names = current_names;
-            self.names.extend(names.iter().cloned());
-            names
-        } else {
-            take(&mut self.names)
-        };
-
         if let Some(directive) = directive {
+            let mut child_names = if self.should_track_names {
+                let names = take(&mut self.names);
+                self.names = current_names;
+                self.names.extend(names.iter().cloned());
+                names
+            } else {
+                take(&mut self.names)
+            };
```

Comment:
side-note: this feels convoluted? it's shuffling and cloning stuff around a lot and i can't rly tell why. but that's pre-existing code, would just be good to clean up

Reply:
Agreed, how about [this](https://github.com/vercel/next.js/pull/73189/commits/7e7de4a318290ed765721c638d807401f685b4f3)?